jax/_src/ops/scatter.py

```diff
@@ -15,11 +15,27 @@
 # Helpers for indexed updates.
 
 
+import sys
+from typing import Any, Optional, Sequence, Tuple, Union
+
+import numpy as np
+
 from jax import lax
 from jax._src.numpy import lax_numpy as jnp
 from jax._src import util
 
 
+Array = Any
+if sys.version_info >= (3, 10):
+    from typing import EllipsisType
+    SingleIndex = Union[None, int, slice, Sequence[int], np.ndarray, EllipsisType]
+else:
+    SingleIndex = Union[None, int, slice, Sequence[int], np.ndarray]
+Index = Union[SingleIndex, Tuple[SingleIndex, ...]]
+Scalar = Union[complex, float, int, np.number]
+Numeric = Union[Array, Scalar]
```

Comment:
Often the right-hand side can be scalar, so I defined an appropriate scalar type.